### PR TITLE
Refactor admin-ui react

### PR DESCRIPTION
**Pull request by matheusps**
#### What is the purpose of this pull request?

A refactor on the `admin-ui-react` package featuring:

- Remove the `@emotion/is-prop-valid` dependency
- Dead code elimination
- Better file organization

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Quality improvement (tests or refactors)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Trivial change (small fix or feature that doesn't impact functionalities)
- [ ] Requires change to documentation, which has been updated accordingly

### CODE REVIEW

File: packages/admin-ui-react/src/system-deprecated/util.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import type { StyleProp } from '@vtex/admin-ui-core'
+
+import { __options, __stylesheet } from './symbols'
+
+export function useOptionsIdentity<Options, Props>(_: Options, props: Props) {
+  return props
+}
+
+/**
+ * get the stylesheet from a AdminUI component
```
Comment (kevinch) [inline packages/admin-ui-react/src/system-deprecated/util.ts:11]: Nitpick
```suggestion
 * get the stylesheet from an Admin UI component
```

File: packages/admin-ui-react/src/system/context.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+/**
+ * Access the System context
+ * @returns
+ */
+export function useSystem(): SystemContextType {
+  const ctx = useContext(SystemContext)
+
+  invariant(
+    ctx,
+    'Something is wrong, make sure you are using the useSystem() hook under an AdminUI provider.'
```
Comment (kevinch) [inline packages/admin-ui-react/src/system/context.ts:22]: `AdminUI` or `Admin UI`?

**Review by kevinch — APPROVED** [review]

File: packages/admin-ui-react/src/system-deprecated/tag.tsx [hunk 3]
```diff
@@ -1,10 +1,10 @@
 import React from 'react'
 import { isFunction, capitalize } from '@vtex/admin-ui-util'
 
-import { useSystem } from './context'
+import { useSystem } from '../system'
```
Comment (lucasaarcoverde) [inline packages/admin-ui-react/src/system-deprecated/tag.tsx:4]: tag is a deprecated feature or only its current implementation?

Comment (matheusps) [reply packages/admin-ui-react/src/system-deprecated/tag.tsx:4]: the current implementation

File: packages/admin-ui-react/src/system/context.ts (hunk 2 above)
Comment (matheusps) [reply packages/admin-ui-react/src/system/context.ts:22]: Admin UI is the correct one, I guess

**Pull request merged**